packages/transport/src/modules/accounts/IdentityDeletionProcessController.ts

```diff
@@ -0,0 +1,114 @@
+    }
+
+    public async initiateIdentityDeletionProcess(): Promise<IdentityDeletionProcess> {
+        const identityDeletionProcessResponse = await this.identityDeletionProcessClient.initiateIdentityDeletionProcess();
+
+        const identityDeletionProcess = this.createIdentityDeletionProcessFromBackboneResponse(identityDeletionProcessResponse);
+
+        await this.identityDeletionProcessCollection.create(identityDeletionProcess);
+        this.eventBus.publish(new IdentityDeletionProcessStatusChangedEvent(this.parent.identity.address.toString(), identityDeletionProcess));
+
+        return identityDeletionProcess;
+    }
+
+    public async cancelIdentityDeletionProcess(identityDeletionProcessId: string): Promise<IdentityDeletionProcess> {
+        const identityDeletionProcessResponse = await this.identityDeletionProcessClient.cancelIdentityDeletionProcess(identityDeletionProcessId);
+        const identityDeletionProcess = this.createIdentityDeletionProcessFromBackboneResponse(identityDeletionProcessResponse);
+        await this.updateIdentityDeletionProcess(identityDeletionProcess);
+        return identityDeletionProcess;
+    }
+
+    public async getIdentityDeletionProcess(identityDeletionProcessId: string): Promise<IdentityDeletionProcess | undefined> {
+        const identityDeletionProcess = await this.identityDeletionProcessCollection.findOne({ id: identityDeletionProcessId });
+        return identityDeletionProcess ? IdentityDeletionProcess.from(identityDeletionProcess) : undefined;
+    }
+
+    public async getIdentityDeletionProcesses(): Promise<IdentityDeletionProcess[]> {
+        return (await this.identityDeletionProcessCollection.find())
+            .map((identityDeletionProcess) => (identityDeletionProcess ? IdentityDeletionProcess.from(identityDeletionProcess) : undefined))
+            .filter((identityDeletionProcess) => !!identityDeletionProcess) as IdentityDeletionProcess[];
+    }
+
+    public async getIdentityDeletionProcessByStatus(
+        identityDeletionProcessStatus: IdentityDeletionProcessStatus | IdentityDeletionProcessStatus[]
+    ): Promise<IdentityDeletionProcess | undefined> {
+        if (!Array.isArray(identityDeletionProcessStatus)) {
+            identityDeletionProcessStatus = [identityDeletionProcessStatus];
+        }
+        const identityDeletionProcess = await this.identityDeletionProcessCollection.findOne({
+            $or: identityDeletionProcessStatus.map((status) => {
+                return { "cache.status": status };
+            })
+        });
+        return identityDeletionProcess ? IdentityDeletionProcess.from(identityDeletionProcess) : undefined;
+    }
+
+    public async fetchCaches(ids: CoreId[]): Promise<{ id: CoreId; cache: CachedIdentityDeletionProcess }[]> {
+        if (ids.length === 0) return [];
+        const getIdentityDeletionPromises = ids.map((id) => this.identityDeletionProcessClient.getIdentityDeletionProcess(id.toString()));
```

Comment:
"getIdentityDeletionProcesses" instead of "getIdentityDeletionPromises"

Reply:
There is no await in the map, so at this point we still have the promises that resolve to the IdentityDeletionProcesses.